cli/crates/server/src/proxy.rs

```diff
@@ -91,32 +93,57 @@ async fn root(State(ProxyState { pathfinder_html, .. }): State<ProxyState>) -> i
     pathfinder_html
 }
 
+const POLL_INTERVAL: Duration = Duration::from_millis(200);
+
 async fn graphql(
     State(ProxyState { client, .. }): State<ProxyState>,
     mut req: Request<Body>,
 ) -> Result<impl IntoResponse, impl IntoResponse> {
     let query = req.uri().query().map_or(String::new(), |query| format!("?{query}"));
 
-    let worker_port = WORKER_PORT.load(Ordering::Relaxed);
+    // http::Request can't be cloned
+    let (parts, body) = req.into_parts();
+    let body_bytes = hyper::body::to_bytes(body).await.map_err(|_| StatusCode::BAD_REQUEST)?;
+    let body = String::from_utf8(body_bytes.clone().into()).map_err(|_| StatusCode::BAD_REQUEST)?;
+    req = Request::from_parts(parts, body_bytes.into());
 
-    if worker_port == 0 {
-        return Err(StatusCode::SERVICE_UNAVAILABLE);
-    }
+    loop {
```

Comment:
can't you rather loop while you have a `0` port? This should avoid all the request stuff as you would use it outside the loop